packages/arb-token-bridge-ui/src/components/TransactionsTable/TransactionsTable.tsx

```diff
@@ -30,87 +45,278 @@ function EmptyTableRow({ children }: { children: React.ReactNode }) {
+      pageParams.pageNumber,
+      pageParams.pageSize,
+      pageParams.searchString
+    ],
+    fetchTransactions,
+    {
+      shouldRetryOnError: false,
+      revalidateIfStale: false,
+      revalidateOnFocus: false,
+      revalidateOnReconnect: false
+    }
+  )
+
+  const disableNextBtn = loading || transactions.length < pageParams.pageSize // if transactions are less than pagesize
+  const disablePrevBtn = loading || !pageParams.pageNumber // if page number is 0, then don't prev.
+
+  const search = () => {
+    const trimmedSearchString = searchString.trim()
+    if (trimmedSearchString && !validate_txhash(trimmedSearchString)) {
+      setSearchError(true)
+      return
+    }
+    // search logic - using `searchString`
+    setPageParams(prevParams => ({
+      ...prevParams,
+      pageNumber: 0,
+      pageSize: 10,
+      searchString: trimmedSearchString
+    }))
+  }
+
+  const next = () => {
+    if (!disableNextBtn) {
+      setPageParams(prevParams => ({
+        ...prevParams,
+        pageNumber: prevParams.pageNumber + 1
+      }))
+    }
+  }
+
+  const prev = () => {
+    if (!disablePrevBtn) {
+      setPageParams(prevParams => ({
+        ...prevParams,
+        pageNumber: prevParams.pageNumber - 1
+      }))
+    }
+  }
+
+  const status = loading
+    ? TableStatus.LOADING
+    : error
+    ? TableStatus.ERROR
+    : TableStatus.SUCCESS
+  const layerType = type === 'deposits' ? 'L1' : 'L2'
 
   return (
-    <table
-      className={`w-full rounded-tr-lg rounded-br-lg rounded-bl-lg bg-gray-1 ${className}`}
-    >
-      <thead className="border-b border-gray-10 text-left text-sm text-gray-10">
-        <tr>
-          <th className="py-3 pl-6 pr-3 font-normal">Status</th>
-          <th className="px-3 py-3 font-normal">Time</th>
-          <th className="px-3 py-3 font-normal">Amount</th>
-          <th className="px-3 py-3 font-normal">TxID</th>
-          <th className="py-3 pl-3 pr-6 font-normal">
-            {/* Empty header text */}
-          </th>
-        </tr>
-      </thead>
-
-      <tbody>
-        {status === 'loading' && (
-          <EmptyTableRow>
-            <div className="flex flex-row items-center space-x-3">
-              <Loader type="TailSpin" color="black" width={16} height={16} />
-              <span className="text-sm font-medium">Loading transactions</span>
-            </div>
-          </EmptyTableRow>
-        )}
-
-        {status === 'error' && (
-          <EmptyTableRow>
-            <span className="text-sm font-medium text-brick-dark">
-              Failed to load transactions
+    <>
+      {/* top search and pagination bar */}
+      <div className="sticky top-0 left-0 flex w-auto flex-nowrap items-center justify-between gap-4 rounded-tr-lg bg-white p-3 text-sm">
+        {/* Search bar */}
+        <div className="relative flex h-full w-full grow items-center rounded border-2 bg-white px-2">
+          <SearchIcon className="h-4 w-4 shrink-0 text-gray-9" />
```

Comment:
a few things
1) search bar should be outside of the table according to mockup, unless the mockup is outdated?
2) the "In the past 14 days...." warning was not implemented on this PR
3) copy on tooltip for expired transactions wasn't updated on this PR

mockup
<img width="774" alt="image" src="https://user-images.githubusercontent.com/13184582/217239642-a95a5796-fa6e-42de-830c-bb08e6a032b1.png">

PR
<img width="707" alt="image" src="https://user-images.githubusercontent.com/13184582/217239866-b1d2adce-639b-4bfa-b1d8-d8fbc8129cb7.png">